src/rgw/rgw_sal_motr.cc

```diff
@@ -2504,6 +2595,63 @@ int MotrAtomicWriter::process(bufferlist&& data, uint64_t offset)
   return this->write();
 }
 
+void MotrObject::update_null_version_index(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
+{
+  int rc = 0;
+  string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
+  string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
+  std::string null_version_key = this->get_name() + "/null";
+  bufferlist::const_iterator iter;
+  bufferlist bl, bl_null_idx_val;
+  rgw_bucket_dir_entry current_null_key;
+  current_null_key.key.name = ent.key.name;
+  current_null_key.key.instance = ent.key.instance;
+  current_null_key.encode(bl_null_idx_val);
+
+  // TODO : cache
+  // GET an extra entry
+  rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                              M0_IC_GET, null_version_key, bl);
+  ldpp_dout(dpp, 20) <<__func__<< "GET null rc : " << rc << dendl;
+
+  if(rc == 0)
+  {
+    rgw_bucket_dir_entry null_key;
+    bufferlist& blr = bl;
+    iter = blr.cbegin();
+    null_key.decode(iter);
+    std::string prev_null_key = null_key.key.name + '[' + null_key.key.instance + ']';
```

Comment:
this code still referring old way to refer null versioned obj key with [] i.e as obj1[null] ? It should be obj1/null

Reply:
Here, we are forming a string to generate previous null object key name (obj[v1], not the null index key - obj/null) to delete it from the motr.